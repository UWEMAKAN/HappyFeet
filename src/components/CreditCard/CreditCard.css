@import url('https://fonts.googleapis.com/css?family=Fjalla+One');

.CreditCard {
  display: flex;
  font-size: 1.6rem;
}

.CreditCard .Form {
  margin-left: 15rem;
}

.CreditCard .Form .CardInfo:first-child {
  display: flex;
  flex-direction: column;
  padding: 2rem;
  border: 1px solid rgba(0, 0, 0, 0.5);
}

.CreditCard .Form .CardInfo span {
  margin-bottom: 1rem;
}

.CreditCard .Form .CardInfo input {
  width: 25rem;
  padding: 0.5rem;
  font-size: 2rem;
}

.CreditCard .Form .CVV_Expiry {
  display: flex;
}

.CreditCard .Form .CVV {
  width: 10rem;
  display: flex;
  flex-direction: column;
}

.CreditCard .Form .CVV input {
  width: 7rem;
}

.CreditCard .Form .Expiry {
  display: flex;
  flex-direction: column;
}

.CreditCard .Form .Expiry input {
  width: 5rem;
}

.CreditCard .Form .Expiry input:first-child {
  margin-right: .5rem;
}

.CreditCard .Form .Expiry input:last-child {
  margin-left: .5rem;
}

.CreditCard .Form input:not(:last-child) {
  margin-bottom: 2rem;
}

.credit-card {
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  width: 50rem;
  height: 31.25rem;
  padding: 2.5rem;
  border-radius: 1.5rem;

  color: white;

  background-image: linear-gradient(25deg, #0f509e, #1399cd);
  box-shadow: 0 0 2rem rgba(0, 0, 0, 0.5);
}

.credit-card__number {
  display: flex;
  justify-content: center;

  font-family: 'Fjalla One', sans-serif;
  font-size: 4.5rem;
}

.credit-card__info {
  display: flex;
  justify-content: space-between;

  font-family: 'Fjalla One', sans-serif;
  font-size: 2.5rem;
}

.credit-card__info_label {
  font-size: 2rem;
}

.credit-card__info_name {
  width: 35rem;
}

.credit-card__info_expiry {
  width: 15rem;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
}

.credit-card__info_cvv {
  width: 10rem;
}

.credit-card__info_name div:last-child,
.credit-card__info_cvv div:last-child,
.credit-card__info_expiry div:last-child {
  height: 2.5rem;
}
