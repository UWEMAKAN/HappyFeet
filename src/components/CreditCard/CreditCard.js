import React, { Component } from 'react';
import './CreditCard.css';

class CreditCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      number: '',
      name: '',
      month: '',
      year: '',
      cvv: ''
    };
  }

  onChangeHandler = (event) => {
    const { name, value } = event.target;
    this.setState((prevState) => ({
      ...prevState, [name]: value
    }));
  }

  render() {
    const {
      number, name, cvv, month, year
    } = this.state;
    const expiry = month.length === 2 ? `${month}/${year}` : month;
    return (
      <div className="CreditCard">
        <div className="credit-card">
          <div className="credit-card__logo">
            <img className="logo" alt="logo" src="http://localhost:3001/assets/images/visa.png" width="60" />
          </div>

          <div className="credit-card__number">{number}</div>

          <div className="credit-card__info">
            <div className="credit-card__info_name">
              <div className="credit-card__info_label">{'CARDHOLDER\'S NAME'}</div>
              <div>{name}</div>
            </div>

            <div className="credit-card__info_cvv">
              <div className="credit-card__info_label">CVV</div>
              <div>{cvv}</div>
            </div>

            <div className="credit-card__info_expiry">
              <div className="credit-card__info_label">EXPIRY</div>
              <div>{expiry}</div>
            </div>
          </div>
        </div>
        <div className="Form">
          <div className="CardInfo">
            <span>{'CARDHOLDER\'S NAME'}</span>
            <input maxLength="20" onChange={this.onChangeHandler} type="text" name="name" />
            <span>CARD NUMBER</span>
            <input maxLength="16" onChange={this.onChangeHandler} type="text" name="number" />
            <div className="CVV_Expiry">
              <div className="CVV">
                <span>CVV</span>
                <input maxLength="3" onChange={this.onChangeHandler} type="text" name="cvv" />
              </div>
              <div className="Expiry">
                <span className="Span">EXPIRY</span>
                <div>
                  <input maxLength="2" onChange={this.onChangeHandler} type="text" name="month" />
                  /
                  <input maxLength="2" onChange={this.onChangeHandler} type="text" name="year" />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default CreditCard;
