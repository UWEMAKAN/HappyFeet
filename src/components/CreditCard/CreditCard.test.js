import React from 'react';
import { shallow } from 'enzyme';
import CreditCard from './CreditCard';

const renderCreditCard = () => shallow(<CreditCard />);

describe('CreditCard Component', () => {
  test('should render CreditCard Component', () => {
    const wrapper = renderCreditCard();
    expect.assertions(1);
    expect(wrapper).toMatchSnapshot();
  });
});
