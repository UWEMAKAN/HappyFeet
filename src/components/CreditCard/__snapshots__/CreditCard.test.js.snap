// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditCard Component should render CreditCard Component 1`] = `ShallowWrapper {}`;
